### PR TITLE
remove unused portProducer parameter and use runtime httpConfiguration in runtime recording

### PR DESCRIPTION
**Pull request by machi1990**
/cc @geoand minor cleanups in VertxHttpProcessor

### CODE REVIEW

File: extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java [hunk 1]
```diff
@@ -126,6 +124,10 @@ ServiceStartBuildItem finalizeRouter(
                 listOfFilters, vertx.getVertx(), router.getRouter(), httpBuildTimeConfig.rootPath, launchMode.getLaunchMode());
 
         boolean startVirtual = requireVirtual.isPresent() || httpConfiguration.virtual;
+        if (requireVirtual.isPresent() || httpConfiguration.virtual) {
```
Comment (geoand) [inline extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java]: Can you revert this change and the relevant removal please?

Comment (machi1990) [reply extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java]: The `HttpConfiguration` class is a runtime configuration, we are trying to access a value from a normal build step (build time), that's what motivated to move that part here (inside a RUNTIME recorder).  Is this okay?

Comment (geoand) [reply extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java]: The fact that the build step is also a recorder doesn't affect the access of build / runtime time configuration.
However your latest addition does make things a little more clear so let's go with it. I'll just add one comment about the name.

File: extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java [hunk 2]
```diff
@@ -99,10 +96,11 @@ VertxWebRouterBuildItem initializeRouter(VertxHttpRecorder recorder,
...
             Optional<RequireVirtualHttpBuildItem> requireVirtual,
-            InternalWebVertxBuildItem vertx,
-            LaunchModeBuildItem launchMode, ShutdownContextBuildItem shutdown, List<DefaultRouteBuildItem> defaultRoutes,
-            List<FilterBuildItem> filters, VertxWebRouterBuildItem router, EventLoopCountBuildItem eventLoopCountBuildItem,
-            HttpBuildTimeConfig httpBuildTimeConfig,
+            InternalWebVertxBuildItem vertx, LaunchModeBuildItem launchMode, ShutdownContextBuildItem shutdown,
+            List<DefaultRouteBuildItem> defaultRoutes, List<FilterBuildItem> filters,
+            VertxWebRouterBuildItem router, EventLoopCountBuildItem eventLoopCountBuildItem,
+            HttpBuildTimeConfig httpBuildTimeConfig, HttpConfiguration httpConfiguration,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClassBuildItemBuildProducer,
```
Comment (geoand) [inline extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/VertxHttpProcessor.java]: Let's be consistent with the names of the parameters. None of the other parameters contains `BuildItem` in them, nor is `Producer` used for producers in other methods of the class.

Sorry for these nitpicks, but I strongly think we should try and be as consistent as possible, especially when it doesn't come at the cost of anything else.

**Review by geoand — APPROVED** [review]
Thanks!

**Comment by machi1990** [comment]
Thanks @geoand

**Comment by geoand** [comment]
Thank you @machi1990
